### PR TITLE
[cherry-picker] Support copying multiple release-notes to cherry-picked PR

### PR DESCRIPTION
**Pull request by oliver-goetz**
/kind enhancement

**What this PR does / why we need it**:
Cherry-picker did not support copying multiple release notes section into the cherry-picked PR, but copied the first one only. This PR adds support to copy multiple release-notes to the cherry-picked PR.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: prow/external-plugins/cherrypicker/server_test.go [hunk 1]
```diff
@@ -1172,6 +1172,16 @@ func TestReleaseNoteFromParentPR(t *testing.T) {
 			input:    "```feature developer #999\nUpdate the magic number from 42 to 49\n```",
 			expected: "```feature developer github.com/foo/bar #123 @foo-author\nUpdate the magic number from 42 to 49\n```",
 		},
+		{
+			name:     "test9",
+			input:    "```feature developer\nUpdate the magic number from 42 to 49\n```\n\n```feature operator\nUpdate another magic number\n```",
```
Comment (plkokanov) [inline prow/external-plugins/cherrypicker/server_test.go:1177]: Small q: I see that there are two newlines between the release notes. Is this intentional for the test? Usually we only separate multiple release notes with 1 new line.

File: prow/external-plugins/cherrypicker/server.go [hunk 2]
```diff
@@ -686,26 +686,33 @@ func normalize(input string) string {
...
-		strings.TrimSpace(potentialMatch[3]),
-		source,
-		ref,
-		author,
-		strings.TrimSpace(potentialMatch[7]),
-	)
+	var output string
+	potentialMatches := releaseNoteRe.FindAllStringSubmatch(body, -1)
+	for i, potentialMatch := range potentialMatches {
+		if potentialMatch == nil {
```
Comment (plkokanov) [inline prow/external-plugins/cherrypicker/server.go:692]: Can we still get a `potentialMatch` that is nil here? If there are no matches I guess `potentialMatches` will be nil. As far as I understand it, if `potentialMatches` is not nil then there should be at least one submatch, namely the `potentialMatch[0]` one.

File: prow/external-plugins/cherrypicker/server.go [hunk 3]
```diff
@@ -686,26 +686,33 @@ func normalize(input string) string {
...
+			return ""
```
Comment (plkokanov) [inline prow/external-plugins/cherrypicker/server.go:693]: If a single `potentialMatch` can be nil, shouldn't we just `continue` here instead of returning?

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR!
I have a few minor comments/questions.

File: prow/external-plugins/cherrypicker/server_test.go (hunk 1 above)
Comment (oliver-goetz) [reply prow/external-plugins/cherrypicker/server_test.go]: I removed the second `\n` 😄

File: prow/external-plugins/cherrypicker/server.go (hunk 2 above)
Comment (oliver-goetz) [reply prow/external-plugins/cherrypicker/server.go]: You are right `potentialMatches == nil` was a special handling when `FindStringSubmatch` was used.

**Pull request merged**